core/crypto/src/test_utils.rs

```diff
@@ -33,6 +34,19 @@ impl PublicKey {
             _ => unimplemented!(),
         }
     }
+
+    pub fn from_implicit_account(account_id: &AccountId) -> Self {
+        assert!(account_id.is_implicit());
+        let mut public_key_data = Vec::with_capacity(33);
+        public_key_data.push(KeyType::ED25519 as u8);
```

Comment:
Parts of this code seem to be taken/copied almost verbatim from [runtime/runtime/src/actions.rs](https://github.com/near/nearcore/blob/99daac9e87cd28e42cf0ce9b9fe228501a4b5170/runtime/runtime/src/actions.rs#L454-L463) so this would perhaps be better off living in a common (non-test-utils) code that either tests or code could use.

(If that's not feasible for any reason, the actions.rs code could be adjusted at least to replace the magic `0` constant with the `KeyType` as is done here, as you're touching the relevant code anyway…)

Reply:
Yeah, we even have another instance of this code copied into near-mirror.


https://github.com/near/nearcore/issues/8588